packages/runtime-core/src/componentEmits.ts

```diff
@@ -113,7 +113,8 @@ export function emit(
   const isModelListener = event.startsWith('update:')
 
   // for v-model update:xxx events, apply modifiers on args
-  const modelArg = isModelListener && event.slice(7)
+  const eventArg = isModelListener && event.slice(7)
+  const modelArg = eventArg && (hyphenate(eventArg) in props ? hyphenate(eventArg) : camelize(eventArg))
```

Comment:
I know it's been nearly two years since this was opened, but I think this PR is still relevant with the latest `main` branch and I have a suggestion.

I believe there's a third option that needs to be handled here, not just camel and kebab case. I think an exact match also needs to be considered.

Here's an example:

* [SFC Playground](https://play.vuejs.org/#eNp9Uk1v2zAM/SuaLk6B1D70FrjBti7DNmBb0Q/0UPWQyrSjVpYEfaQpXP/3UlKSFkWRiyHyPZKP5hvoN2PKdQA6o7XjVhhPHPhg5kyJ3mjryUAstGQkrdU9KZBa7KG/z2crIZstVFbbOPZDElNcK+dJ7zpyGptMil8gpSY32srmS3HEVF3lmTgNAw+9kUsPGBFSGwvzl2FI5eP4UlcxkZDd2PVxrxuQs+U9P/6x+Fl6K/pTRrGAUVIht672LemUeod6WtGVD04r3HeIzRjlujdCgv1vvEC9jM5IQiK2RLlPf1LO2wDTXZ6vgD9+kn9wm5hj9NyCA7sGRveYX9oOfIYXl/9gg+89iIsEiewD4AU4LUPUmGnfg2pQ9jteUvs7nUao7sotNh6U2y0VhUbmmPiM4pHODqz+JvekPEl1TI34F9/d+BPPNNAKBedWGze5LbaHKe7w1NkL0AuPZsi0BQaRFkyDJ9qdMbGZaoPiURXJ6OQoS4sNJh8rpqRAzd1K4DfaCqUectZ98B4bf+VS8Ec0TO7G6Pw6PeoqEz4YaHwFFOAQXw==)

While I accept it is not common practice, it is possible for the `v-model` argument to be written in neither camel case nor kebab case. In the example above I use `v-model:abc-DEF.trim`. This does work with the current code on `main`, but it no longer works with the change proposed in this PR.

Even though it's an edge case, I think it's important that an exact match should always work. Case coercion logic shouldn't change that.

There's similar logic further down the same file for determining the value of `handlerName`. That tries an exact match first, before considering any case changes. I think the logic for modifiers should be consistent with that.

Reply:
I've opened #9609, which builds on the work done here and also covers the case I mentioned above.